Armadillo/src/main/java/com/scribd/armadillo/encryption/SecureStorage.kt

```diff
@@ -1,44 +1,94 @@
         const val ALLOWED_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXTZabcdefghiklmnopqrstuvwxyz0123456789"
         const val DEFAULT = "82YEDKqPBEqA2qAb4bUU"
-        const val LOCATION = "armadillo.storage"
+        const val STANDARD_STORAGE_FILENAME = "armadillo.storage"
+        const val DOWNLOAD_FILENAME = "armadillo.download"
         const val TAG = "SecureStorage"
     }
 
     override fun downloadSecretKey(context: Context): ByteArray {
-            val sharedPreferences = context.getSharedPreferences(LOCATION, Context.MODE_PRIVATE)
-            return if (sharedPreferences.contains(DOWNLOAD_KEY)) {
-                val storedKey = sharedPreferences.getString(DOWNLOAD_KEY, DEFAULT)!!
-                if(storedKey == DEFAULT){
-                    Log.e(TAG, "Storage Is Out of Alignment")
-                }
-                storedKey.toSecretByteArray
-            } else {
-                createRandomString().also {
-                    sharedPreferences.edit().putString(DOWNLOAD_KEY, it).apply()
-                }.toSecretByteArray
+        val sharedPreferences = context.getSharedPreferences(STANDARD_STORAGE_FILENAME, Context.MODE_PRIVATE)
+        return if (sharedPreferences.contains(DOWNLOAD_KEY)) {
+            val storedKey = sharedPreferences.getString(DOWNLOAD_KEY, DEFAULT)!!
+            if (storedKey == DEFAULT) {
+                Log.e(TAG, "Storage Is Out of Alignment")
             }
+            storedKey.toSecretByteArray
+        } else {
+            createRandomString().also {
+                sharedPreferences.edit().putString(DOWNLOAD_KEY, it).apply()
+            }.toSecretByteArray
         }
+    }
 
     private fun createRandomString(): String {
-            return (1..STRING_LENGTH)
-                    .map { ALLOWED_CHARS.random() }
-                    .joinToString("")
+        return (1..STRING_LENGTH)
+            .map { ALLOWED_CHARS.random() }
+            .joinToString("")
+    }
+
+    override fun saveDrmDownload(context: Context, audioUrl: String, drmDownload: DrmDownload) {
+        context.getSharedPreferences(DOWNLOAD_FILENAME, Context.MODE_PRIVATE).also { sharedPrefs ->
+            val key = getDrmDownloadKey(audioUrl, drmDownload.drmType)
+            val value = Base64.encodeToString(Json.encodeToString(drmDownload).toByteArray(StandardCharsets.UTF_8), Base64.NO_WRAP)
+            sharedPrefs.edit().putString(key, value).apply()
+        }
+    }
+
+    override fun getDrmDownload(context: Context, audioUrl: String, drmType: DrmType): DrmDownload? =
+        context.getSharedPreferences(DOWNLOAD_FILENAME, Context.MODE_PRIVATE).let { sharedPrefs ->
+            sharedPrefs.getString(getDrmDownloadKey(audioUrl, drmType), null)?.decodeToDrmDownload()
+        }
+
+    override fun getAllDrmDownloads(context: Context): Map<String, DrmDownload> =
+        context.getSharedPreferences(DOWNLOAD_FILENAME, Context.MODE_PRIVATE).let { sharedPrefs ->
+            sharedPrefs.all.keys.mapNotNull { key ->
```

Comment:
Do you need to skip `DOWNLOAD_KEY`?

Reply:
I've put all the DRM download prefs into its own shared prefs file, so we don't need to exclude anything